x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -207,9 +208,9 @@ private static void checkInvalidNamedExpressionUsage(
         } else if (e.foldable()) {
             // don't do anything
         } else if (groups.contains(e) || groupRefs.contains(e)) {
-            // if (level != 0) {
-            // failures.add(fail(e, "scalar functions over groupings [{}] not allowed yet", e.sourceText()));
-            // }
+            if (level == 0) {
+                failures.add(fail(e, "remove grouping key [{}] as is already included in the STATS output", e.sourceText()));
+            }
```

Comment:
Do we need this, as it seems to be enforced in the `LogicalPlanBuilder` already?

Commenting out these lines still makes all verifier and csv tests pass.